### PR TITLE
Add a way to run multithreaded lookups in mmdblookup

### PR DESCRIPTION
**Pull request by horgh**
This is similar to the existing benchmark mode, but runs in several
threads.

### CODE REVIEW

**Comment by horgh** [comment]
Not too certain about the Windows detection here.

Also, perhaps this would be better as a separate program, or be combined more with the existing benchmark code. Possibly it doesn't belong in this repo at all. I did find it useful for finding one race, so I think it or something like it would be good to have somewhere.

File: bin/mmdblookup.c [hunk 1]
```diff
@@ -131,13 +164,14 @@ LOCAL const char **get_options(int argc, char **argv, char **mmdb_file,
             { "verbose",   no_argument,       0, 'v' },
             { "version",   no_argument,       0, 'n' },
             { "benchmark", required_argument, 0, 'b' },
+            { "threads",   required_argument, 0, 't' },
```
Comment (oschwald) [inline bin/mmdblookup.c]: Could we remove this option on Windows?

File: bin/mmdblookup.c [hunk 2]
```diff
@@ -396,3 +431,105 @@ LOCAL void random_ipv4(char *ip)
...
+};
+#endif
+
+LOCAL bool start_threaded_benchmark(
+    MMDB_s *const mmdb,
+    int const thread_count,
+    int const iterations)
+{
+#ifdef _WIN32
+    return false;
```
Comment (oschwald) [inline bin/mmdblookup.c]: Why not just remove the function definitions for these on Windows altogether and remove the following?
 https://github.com/maxmind/libmaxminddb/blob/ec604d71879583604d8efeae2d288474d710bf01/bin/mmdblookup.c#L82-L89

File: bin/mmdblookup.c (hunk 1 above)
Comment (horgh) [reply bin/mmdblookup.c]: Good idea.

File: bin/mmdblookup.c (hunk 2 above)
Comment (horgh) [reply bin/mmdblookup.c]: Yeah, that's nicer. Mainly I was trying to work around `regen-prototypes.pl` not being happy with macros (it deletes them from its section). I found a different workaround that seems better, although is still a bit ugly what with the extra `NO_PROTOTYPE` macro. I wonder if we need `regen-prototypes.pl`. It's not a huge pain to keep them up to date manually and would avoid this problem.

Comment (oschwald) [reply bin/mmdblookup.c]: Yeah, I would be fine getting rid of that script. I don't find it particularly useful.

**Pull request merged**